src/components/TokenEnterAmount.tsx

```diff
@@ -35,6 +41,200 @@ function groupNumber(value: string) {
+  token: TokenBalance
+  onSelectToken?: (token: TokenBalance) => void
+}) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  const inputRef = useRef<RNTextInput>(null)
+  const bottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const [amount, setAmount] = useState('')
+  const [amountType, setAmountType] = useState<AmountEnteredIn>('token')
+
+  // this should never be null, just adding a default to make TS happy
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const tokenInfo = useTokenInfo(props.token.tokenId)
+  const usdToLocalRate = useSelector(usdToLocalCurrencyRateSelector)
+
+  const amountRaw = useMemo(() => {
+    return amount.replaceAll(groupingSeparator, '').replaceAll(decimalSeparator, '.')
+  }, [amount])
+
+  const derived = useMemo(() => {
+    if (amountType === 'token') {
+      const parsedTokenAmount = parseInputAmount(amount, decimalSeparator)
+      const tokenToLocal = convertTokenToLocalAmount({
+        tokenAmount: parsedTokenAmount,
+        tokenInfo,
+        usdToLocalRate,
+      })
+      const convertedTokenToLocal =
+        tokenToLocal && tokenToLocal.gt(0)
+          ? tokenToLocal.toString().replaceAll('.', decimalSeparator)
+          : ''
+      const parsedLocalAmount = parseInputAmount(
+        convertedTokenToLocal.replaceAll(groupingSeparator, ''),
+        decimalSeparator
+      )
+
+      return {
+        token: {
+          amount: amount,
+          bignum: parsedTokenAmount,
+          readable: roundTokenAmount(amount, props.token),
+        },
+        local: {
+          amount: convertedTokenToLocal,
+          bignum: parsedLocalAmount,
+          readable: roundLocalAmount(convertedTokenToLocal, localCurrencySymbol),
+        },
+      }
+    }
+
+    const parsedLocalAmount = parseInputAmount(
+      amount.replaceAll(groupingSeparator, ''),
+      decimalSeparator
+    )
+
+    const localToToken = convertLocalToTokenAmount({
+      localAmount: parsedLocalAmount,
+      tokenInfo,
+      usdToLocalRate,
+    })
+    const convertedLocalToToken =
+      localToToken && localToToken.gt(0)
+        ? // no group separator for token amount, round to token.decimals and strip trailing zeros
+          localToToken
+            .toFormat(props.token.decimals, { decimalSeparator })
+            .replace(new RegExp(`[${decimalSeparator}]?0+$`), '')
+        : ''
+
+    const parsedTokenAmount = parseInputAmount(convertedLocalToToken, decimalSeparator)
+
+    return {
+      token: {
+        amount: convertedLocalToToken,
+        bignum: parsedTokenAmount,
+        readable: roundTokenAmount(convertedLocalToToken, props.token),
+      },
+      local: {
+        amount: amount,
+        bignum: parsedLocalAmount,
+        readable: roundLocalAmount(amount, localCurrencySymbol),
+      },
+    }
+  }, [amount, amountType, localCurrencySymbol])
+
+  function onOpenTokenPicker() {
```

Comment:
This and all the following handler functions are the copy of the same functions from the EnterAmount/EarnEnterAmount/SwapScreen. The only intention is to have a reusable same handler functions in a single place so it is easier to re-use.

Reply:
@kathaypacific You're right! Following the idea of reducing amount of actions needed to use the hook, I initially though I would be able to make it re-usable similar to how it was done with [onSelectToken](https://github.com/valora-inc/wallet/pull/6243/files#diff-4cca836e5b4816f0318151a8b244d9b9388c799e08403939fcecc59c5b47ad80R217-R221) but while working more on the `SwapScreen` I've realised it would be too cumbersome to try to abstract this part as well.

I'll move this handler back to the component level!